compiler/x/codegen/CompareAnalyser.cpp

```diff
@@ -144,17 +144,28 @@ void TR_X86CompareAnalyser::integerCompareAnalyser(
       }
 
    if (realFirstChild)
-      cg()->recursivelyDecReferenceCount(realFirstChild);
-   else
-      cg()->decReferenceCount(firstChild);
+      {
+	if (!getCmpMem1Reg2()) {
+	   cg()->recursivelyDecReferenceCount(realFirstChild);
+	} else {
+	   cg()->decReferenceCount(realFirstChild);
+	} else {
```

Comment:
I don't think this is right.

Reply:
When fixing this can we take the opportunity to add a proper commit message as per the PR description?